pool/client.go

```diff
@@ -146,39 +146,38 @@ type Client struct {
 	wg          sync.WaitGroup
 }
 
-// generateExtraNonce1 generates a random 4-byte extraNonce1
-// for the client.
-func (c *Client) generateExtraNonce1() error {
+// generateExtraNonce1 generates a random 4-byte extraNonce1 for the client.
+func generateExtraNonce1() (string, error) {
```

Comment:
I don't think this needs to be a func, it isn't reused anywhere. The logic can just be moved into `NewClient` and explained with a comment.

Reply:
Good idea.  Done.